# Case Study 1: From Exploration to Design

- Name: Thanh Tuyen (Anna) Nguyen
- Student ID: 70090352
- Link to GitHub Repository: https://github.com/annanguyen0605/DSCI200_Case_study_1.git

This blank notebook is provided for you to use to answer questions from [Case Study 1](https://ubc-dsci.github.io/dsci-200/case-studies/case-study-1.html). 

In [11]:
install.packages('tidyverse')
devtools::install_github("lbelzile/hecedsm")

Updating HTML index of packages in '.Library'

Making 'packages.html' ...
 done




── R CMD build ─────────────────────────────────────────────────────────────────
* checking for file ‘/tmp/RtmptN3wsC/remotesba1cd6201e/lbelzile-hecedsm-df0a544/DESCRIPTION’ ... OK
* preparing ‘hecedsm’:
* checking DESCRIPTION meta-information ... OK
* checking for LF line-endings in source and make files and shell scripts
* checking for empty or unneeded directories
  NB: this package now depends on R (>= 4.1.0)
  pipe |> or function shorthand \(...) syntax added in R 4.1.0.
  File(s) using such syntax:
    ‘SKD23_S2A.Rd’
* building ‘hecedsm_2025.06.18.tar.gz’



In [12]:
library(tidyverse)
library(hecedsm)

── Attaching core tidyverse packages ──────────────────────── tidyverse 2.0.0 ──
✔ dplyr     1.1.4     ✔ readr     2.1.6
✔ forcats   1.0.1     ✔ stringr   1.6.0
✔ ggplot2   4.0.1     ✔ tibble    3.3.0
✔ lubridate 1.9.4     ✔ tidyr     1.3.1
✔ purrr     1.2.0     
── Conflicts ────────────────────────────────────────── tidyverse_conflicts() ──
✖ dplyr::filter() masks stats::filter()
✖ dplyr::lag()    masks stats::lag()
ℹ Use the conflicted package (<http://conflicted.r-lib.org/>) to force all conflicts to become errors

Attaching package: ‘hecedsm’


The following object is masked from ‘package:stringr’:

    words




# I. Choosing Dataset

## 1. Load data (Dataset Chosen: BL22_E)

In [13]:
data <- BL22_E
head(data)
summary(data)

id writer cond  partner_time task_time room_time
1 1   TRUE  video  84.626      193.868   20.026   
2 1  FALSE  video 122.026      144.806   33.150   
3 2   TRUE  video  86.122      125.630   88.740   
4 2  FALSE  video  89.828      116.518   93.500   
5 3  FALSE  f2f     1.632      255.816   41.480   
6 4  FALSE  video 118.694      160.956   17.578

       id        writer           cond      partner_time       task_time      
 1      :  2   Mode :logical   f2f  :126   Min.   :  0.068   Min.   :  1.156  
 2      :  2   FALSE:131       video:144   1st Qu.: 25.968   1st Qu.:142.222  
 4      :  2   TRUE :139                   Median : 53.975   Median :189.550  
 5      :  2                               Mean   : 72.842   Mean   :181.094  
 6      :  2                               3rd Qu.:111.987   3rd Qu.:232.704  
 7      :  2                               Max.   :287.062   Max.   :294.780  
 (Other):258                                                                  
   room_time      
 Min.   :  0.102  
 1st Qu.: 13.974  
 Median : 31.909  
 Mean   : 45.751  
 3rd Qu.: 69.606  
 Max.   :254.150  
                  

The primary objective of this study was to examine how virtual communication via videoconferencing affects collaborative creative idea generation compared to in-person interaction. Motivated by the rapid expansion of remote work accelerated by COVID-19, the authors sought to understand whether replacing face to face collaboration with virtual meetings imposes cognitive costs on innovation, which is foundational to scientific and commercial progress. While prior research focused on how communication technologies influence information exchange, this study aimed to isolate the effects of the physical nature of interaction itself.

The study is important because organizations increasingly rely on virtual collaboration, yet creativity is critical for problem-solving, research, and competitive advantage. Understanding whether and how videoconferencing alters creative processes has direct implications for workplace design, team management, and innovation strategy.

The authors hypothesized that videoconferencing would reduce the number of creative ideas generated, but not necessarily impair the ability to evaluate or select ideas. Through laboratory and field experiments across five countries, they tested whether virtual interaction narrows cognitive focus, predicting that this constrained attention would suppress idea generation while leaving evaluative decision-making relatively unaffected.

# II. Data exploration

In [14]:
head(data)
glimpse(data)

id writer cond  partner_time task_time room_time
1 1   TRUE  video  84.626      193.868   20.026   
2 1  FALSE  video 122.026      144.806   33.150   
3 2   TRUE  video  86.122      125.630   88.740   
4 2  FALSE  video  89.828      116.518   93.500   
5 3  FALSE  f2f     1.632      255.816   41.480   
6 4  FALSE  video 118.694      160.956   17.578

Rows: 270
Columns: 6
$ id           <fct> 1, 1, 2, 2, 3, 4, 4, 5, 5, 6, 6, 7, 7, 8, 8, 9, 9, 10, 10…
$ writer       <lgl> TRUE, FALSE, TRUE, FALSE, FALSE, FALSE, TRUE, TRUE, FALSE…
$ cond         <fct> video, video, video, video, f2f, video, video, video, vid…
$ partner_time <dbl> 84.626, 122.026, 86.122, 89.828, 1.632, 118.694, 24.684, …
$ task_time    <dbl> 193.868, 144.806, 125.630, 116.518, 255.816, 160.956, 245…
$ room_time    <dbl> 20.026, 33.150, 88.740, 93.500, 41.480, 17.578, 21.828, 3…


- There are 6 variables and 270*6 = 1620 observations.
  
- Categorical variables:

  + id (factor): identifies each participant or pair.

  + cond (factor): represents the communication condition (e.g., video vs. face-to-face).

  + writer (logical): a binary categorical variable indicating whether the participant was the writer (TRUE or FALSE).

- Numerical variables:

  + partner_time (double): time spent interacting with a partner.

  + task_time (double): time spent on the task.

  + room_time (double): time spent in the room.

The numerical variables (`partner_time`, `task_time`, and `room_time`) are already stored as continuous numeric values and are suitable for further exploration of the data. Convert `writer` to a factor instead of logical binary. This ensures it is treated as a categorical predictor in analyses such as group comparisons or regression.

In [17]:
data$writer <- as.factor(data$writer)

In [23]:
colSums(is.na(data))
colMeans(is.na(data)) * 100

id       writer         cond partner_time    task_time    room_time 
           0            0            0            0            0            0

id       writer         cond partner_time    task_time    room_time 
           0            0            0            0            0            0

There is no missing value (NA values) for all of the variables in the dataset. Therefore, 0% of the all variables in the dataset is missing values.

In [29]:
summary(data)

       id        writer       cond      partner_time       task_time      
 1      :  2   FALSE:131   f2f  :126   Min.   :  0.068   Min.   :  1.156  
 2      :  2   TRUE :139   video:144   1st Qu.: 25.968   1st Qu.:142.222  
 4      :  2                           Median : 53.975   Median :189.550  
 5      :  2                           Mean   : 72.842   Mean   :181.094  
 6      :  2                           3rd Qu.:111.987   3rd Qu.:232.704  
 7      :  2                           Max.   :287.062   Max.   :294.780  
 (Other):258                                                              
   room_time      
 Min.   :  0.102  
 1st Qu.: 13.974  
 Median : 31.909  
 Mean   : 45.751  
 3rd Qu.: 69.606  
 Max.   :254.150  
                  

Both the categorical variables `cond` `writer` both have the same amount of observations/valuea (both have 270 observations), thus there is no missing value for any of the variable in the dataset, which means there is no class imbalance between the two categorical variables `cond` and `writer`.